src/go/gcsrunner/serviceaccount.go

```diff
@@ -0,0 +1,87 @@
+package gcsrunner
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+)
+
+// staticTokenSource implements oath2.TokenSource
+type staticTokenSource struct {
+	accessToken string    `json:"access_token"`
+	tokenType   string    `json:"token_type,omitempty"`
+	expiry      time.Time `json:"expiry,omitempty"`
+}
+
+func (s staticTokenSource) Token() (*oauth2.Token, error) {
+	return &oauth2.Token{
+		AccessToken: s.accessToken,
+		Expiry:      s.expiry,
+	}, nil
+}
+
+type iamcredentialsResponse struct {
```

Comment:
Optional: we could also use https://pkg.go.dev/google.golang.org/api/iamcredentials/v1#GenerateAccessTokenResponse here instead of rolling our own response struct.

Reply:
The struct here versus in the iamcredentials API is functionally identical anyway, so I don't think we gain much by including this dependency. The context here is that my attempts to use the iamcredentials API resulted in odd errors I couldn't debug, and when I stepped away from using the API at all, everything just worked™.